src/widgets/RealtimeEffectPanel.cpp

```diff
@@ -0,0 +1,711 @@
+
+            effectPlugin->DoEffect(settings,
+               rate,
+               &*mTrack->GetOwner(),
+               &trackFactory,
+               selectedRegion,
+               flags,
+               &window,
+               EffectUI::DialogFactory,
+               access);
+         });
+      }
+
+      void OnChangeButtonClicked(wxCommandEvent& event)
+      {
+         if(mEffectState == nullptr)
+            return;//not initialized
+
+         ShowSelectEffectMenu(mChangeButton, this);
+      }
+
+      void OnEnableButtonClicked(wxCommandEvent&)
+      {
+         if(mEffectState == nullptr)
+            return;//not initialized
+
+         //TODO: implement
+      }
+
+      void OnPaint(wxPaintEvent&)
+      {
+#ifdef __WXMSW__
+         wxBufferedPaintDC dc(this);
+#else
+         wxPaintDC dc(this);
+#endif
+         const auto rect = wxRect(GetSize());
+
+         dc.SetPen(*wxTRANSPARENT_PEN);
+         dc.SetBrush(GetBackgroundColour());
+         dc.DrawRectangle(rect);
+
+         dc.SetPen(theTheme.Colour(clrEffectListItemBorder));
+         dc.SetBrush(theTheme.Colour(clrEffectListItemBorder));
+         dc.DrawLine(rect.GetBottomLeft(), rect.GetBottomRight());
+      }
+      
+   };
+
+   wxString ShowSelectEffectMenu(wxWindow* parent, RealtimeEffectControl* currentEffectControl)
+   {
+      wxMenu menu;
+
+      if(currentEffectControl != nullptr)
+      {
+         menu.Append(wxID_REMOVE, XO("No Effect").Translation());
+         menu.AppendSeparator();
+      }
+
+      auto& pluginManager = PluginManager::Get();
+
+      std::vector<const PluginDescriptor*> effects;
+      int selectedEffectIndex = -1;
+
+      auto compareEffects = [](const PluginDescriptor* a, const PluginDescriptor* b)
+      {
+         return a->GetVendor() < b->GetVendor()
+            || (a->GetVendor() == b->GetVendor() && a->GetSymbol().Translation() < b->GetSymbol().Translation());
+      };
+
+      for(auto& effect : pluginManager.EffectsOfType(EffectTypeProcess))
+      {
+         if(!effect.IsEffectRealtime() || !effect.IsEnabled())
+            continue;
+
+         auto it = std::lower_bound(effects.begin(), effects.end(), &effect, compareEffects);
+         effects.insert(it, &effect);
+      }
+      
+      wxString currentSubMenuName;
+      std::unique_ptr<wxMenu> currentSubMenu;
+
+      auto submenuEventHandler = [&](wxCommandEvent& event)
+      {
+         selectedEffectIndex = event.GetId() - wxID_HIGHEST; 
+      };
+
+      for(int i = 0, count = effects.size(); i < count; ++i)
+      {
+         auto& effect = *effects[i];
+
+         if(currentSubMenuName != effect.GetVendor())
+         {
+            if(currentSubMenu)
+            {
+               currentSubMenu->Bind(wxEVT_MENU, submenuEventHandler);
+               menu.AppendSubMenu(currentSubMenu.release(), currentSubMenuName);
+            }
+            currentSubMenuName = effect.GetVendor();
+            currentSubMenu = std::make_unique<wxMenu>();
```

Comment:
I imagine different sorting and grouping criteria might be future demands.  Do what now?  Nothing but think about it